yamlfmt.go

```diff
@@ -88,10 +90,12 @@ func (s LineBreakStyle) Separator() (string, error) {
 }
 
 type FeatureFunc func([]byte) ([]byte, error)
+type YAMLFeatureFunc func(yaml.Node) error
```

Comment:
Thanks for your feedback! Keeping `yaml.v3`isms out of the yamlfmt API sounds like a good idea.

I don't think I like the idea of changing the signature to take an `any`, since it will still only be passed a `yaml.Node` (for now at least), and this would move a compile-time check into a documentation-only type hint.

I've pushed a change to move this into the basic formatter, let me know what you think.